### DISCUSSION TITLE
[v2] Does v2 support desktop?

### DISCUSSION THREAD

**Discussion by shapled** [fsprojects/Fabulous#938] [Q&A]

**Comment by TimLariviere** [comment]
Fabulous v2 main target is Xamarin.Forms, so we do support all platforms targeted by Xamarin.Forms including desktop via WPF, UWP or macOS.

**Comment by twop** [comment]
I started to use Fabulous because I wanted to build macOS app. While Fabulous (v1) is a very pleasant experience, the underlying XamarinForms technology just doesn't support macOS very well, I had endless stream of bugs and unresolvable issues.

So I decided to pivot to an iPad app instead, and defer the macOS version until MAUI or maybe to something else entirely.

So in my personal experience macOS is not really a good target for Fabulous / XF, I have no experience with Windows, I assume it should be much better.

**Reply by rastreus** [reply]
@praeclarum has been doing a lot of work for the `net6.0-maccatalyst` [Target framework](https://docs.microsoft.com/en-us/dotnet/standard/frameworks).

[Mac Catalyst](https://developer.apple.com/mac-catalyst/) uses iOS UIKit to bring iPad apps to macOS. I haven't tried it, but that might be a way to bring your Fabulous / XF iPad app to macOS.

**Reply by twop** [reply]
I don't think he finished the effort though for XF :( I hope that I'm wrong about it, would love to use it for my app!

Also, in the lights of MAUI there is probably going to be no chance that XF will get an official support for catalyst (MAUI does support catalyst).

**Reply by rastreus** [reply]
I see this: https://github.com/xamarin/xamarin-macios/wiki/Mac-Catalyst-(Early-Preview), but I'm out of the loop regarding XF and MAUI.

@rolfbjarne Is Mac Catalyst support shipping for Xamarin.Forms or only for MAUI?

[EDIT] It looks like this shipped 3 days ago: https://github.com/xamarin/xamarin-macios/wiki/.NET-release-notes-Xcode-13.3#mac-catalyst

**Reply by twop** [reply]
@TimLariviere this is potentially a big deal for us if xamarin now supports catalyst target

**Reply by TimLariviere** [reply]
This is only for Xamarin not for XF, though?

Xamarin.Forms has entered bug fixes only mode since reaching v5.0.
So it is extremely unlikely it would officially support Mac Catalyst.

I'm sure you could hack support without too much difficulties, like I did with XF on .NET 6.0 (https://github.com/fabulousfx/FSharp.Mobile.Templates/tree/main/templates/Xamarin.Forms).
But it's a PITA because they added breaking changes to Xamarin.iOS and it requires us to build a special version of Xamarin.Forms (https://github.com/fabulousfx/Xamarin.Forms). I guess it would be the same if you want to target `net6.0-maccatalyst`.

**Comment by rolfbjarne** [comment]
@rastreus we won't be adding Mac Catalyst support to the old Xamarin products, it will be .NET only. This means that Xamarin.Forms won't be able to support it either, unless Xamarin.Forms is ported to .NET by someone (in which case I believe it should be fairly trivial, although I'm sure there will be some problems along the way).

**Reply by rastreus** [reply]
Thank you for the clarification.
